lib/Sabberworm/CSS/Parser.php

```diff
@@ -64,20 +71,91 @@ public function __construct($sText, Settings $oParserSettings = null, $iLineNo =
 			$this->aSizeUnits[$iSize][strtolower($val)] = $val;
 		}
 		ksort($this->aSizeUnits, SORT_NUMERIC);
-	}
+		$this->fixCharset();
+	}
+
+	private function fixCharset() {
+		// We need to know the charset before the parsing starts,
+		// UTF BOMs have the highest precedence and must ve removed before other processing.
+		$this->sOriginalCharset = strtolower($this->oParserSettings->sDefaultCharset);
+		if (strpos($this->sText, self::BOM8) === 0) {
```

Comment:
`strpos` searches the whole string for the BOM. I think we should use `strpos(substr($this->sText, 0, strlen(self::BOM8)), self::BOM8) === 0` for performance. Same for all other checks.